### ISSUE TITLE
[SCSS] nested properties with shorthand values break syntax highlighting 

### ISSUE THREAD

**Issue by burtek** [microsoft/vscode#81038]
Issue Type: <b>Bug</b>

[Nested properties of SCSS](https://sass-lang.com/documentation/style-rules/declarations#nesting) break syntax highlighting when provided both shorthand-value and sub-values:

![image](https://user-images.githubusercontent.com/2940958/65031612-d455b880-d941-11e9-8e80-f5628afdad2b.png)
![image](https://user-images.githubusercontent.com/2940958/65032031-a624a880-d942-11e9-96a1-eae09496f9b8.png)

The issue is reproducible with all extensions disabled. Snippet from screenshot:

```scss
.button {
    @include fontDemi;

    border: 0 {
        radius: 30px;
    };
    background-color: $buttonColor;
    font-size: 1vw;
    padding: 1.8vh 1.5vw;
    text-transform: uppercase;
    letter-spacing: 1px;
}
```

VS Code version: Code 1.38.1 (b37e54c98e1a74ba89e03073e5a3761284e3ffb0, 2019-09-11T13:35:15.005Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz (8 x 3498)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.69GB (2.83GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (53)</summary>

Extension|Author (truncated)|Version
---|---|---
replacerules|bhu|0.2.7
vscode-intelephense-client|bme|1.2.3
path-intellisense|chr|1.4.2
bracket-pair-colorizer-2|Coe|0.0.28
vscode-eslint|dba|1.9.1
xml|Dot|2.5.0
eslint-disable-snippets|drK|1.2.0
gitlens|eam|9.9.3
prettier-vscode|esb|2.2.2
vscode-commands|fab|1.2.5
vscode-todo-plus|fab|4.14.1
cshtml-tm|fir|1.0.0
auto-close-tag|for|0.5.6
auto-rename-tag|for|0.1.0
vscode-yarn|gam|1.6.0
sass-lint|gle|1.0.6
gc-excelviewer|Gra|2.1.32
npm-dependency-links|her|1.0.1
beautify|Hoo|1.5.0
vscode-react-typescript|inf|1.3.1
run-in-terminal|kor|0.0.1
yarn-lock-syntax|mar|0.1.3
twig-language|mbl|0.8.10
twig-language-2|mbl|0.8.10
rainbow-csv|mec|1.3.1
code-beautifier|mic|2.3.3
vscode-clang|mit|0.2.3
vscode-apache|mrm|1.2.0
vscode-jade-snippets|mrm|1.0.1
vscode-json5|mrm|1.0.0
vscode-scss|mrm|0.6.2
vscode-docker|ms-|0.8.1
remote-wsl|ms-|0.39.5
csharp|ms-|1.21.2
powershell|ms-|2019.5.0
sublime-keybindings|ms-|4.0.3
typescript-javascript-grammar|ms-|0.0.52
vscode-typescript-tslint-plugin|ms-|1.2.2
debugger-for-chrome|msj|4.11.7
vscode-react-native|msj|0.11.0
color-highlight|nau|2.3.0
vscode-versionlens|pfl|0.24.0
vscode-react-refactor|pla|1.0.5
quicktype|qui|12.0.46
vscode-sort-json|ric|1.14.0
multi-command|ryu|1.3.0
react-component-refactor|san|0.2.0
stylelint|shi|0.51.0
code-spell-checker|str|1.7.18
shell-launcher|Tyr|0.3.0
vscode-icons|vsc|9.3.0
vscode-todo-highlight|way|1.0.4
txt-syntax|xsh|0.0.3

</details>

**Comment by burtek** [comment]
Found out this is duplicate of #46753 and has related issue in upstream repo: https://github.com/atom/language-sass/issues/250

**Comment by burtek** [comment]
That repo is dead :(

**Comment by b5414** [comment]
My [issue with scss highlighting](https://github.com/microsoft/vscode/issues/157480), bug/test code:
```css
.test {
	> input:placeholder-shown + .pupa {
		background: red;
	}

	> input:not(:placeholder-shown) + .lupe {
		background: red;
	}
}
```

**Comment by alexr00** [comment]
Tried out https://github.com/TheRealSyler/vscode-sass-indented, but it is for sass, not scss, and results in significant differences from the current grammar.

**Comment by aeschli** [comment]
The wrong highlighting of the given example is because of the nested rule (`border...`).
![image](https://user-images.githubusercontent.com/2940958/65031612-d455b880-d941-11e9-8e80-f5628afdad2b.png)

`@layer` by itself is well supported:
![image](https://github.com/microsoft/vscode/assets/6461412/f1bca2f5-4bc3-49a5-aa30-d37579f446b0)
```
@layer module, state;

@layer state {
  .alert {
    background-color: brown;
  }
  p {
    border: medium solid limegreen;
  }
}

@layer module {
  .alert {
    border: medium solid violet;
    background-color: yellow;
    color: white;
  }
}
```

**Comment by simonpai** [comment]
Are we still using the old grammar file from that archived repo?

I have a wild idea: can we expose and make the grammar file editable in VSCode settings? And then perhaps we can even have the community contributing back by sharing what we've added in our own settings?

**Comment by alexr00** [comment]
Yes, but we're planning to start working on tree-sitter support in May and we can see if https://github.com/serenadeai/tree-sitter-scss is any better than the current textmate grammar.

**Comment by burtek** [comment]
That repo seems dead as well, I worry we're gonna end up with broken syntax handling again soon...
